### ISSUE TITLE
Compile error with Qt 5.15.2 and recent GCC/LLVM

### ISSUE THREAD

**Issue by garybuhrmaster** [MythTV/mythtv#631]
Compile error with Qt 5.15.2 and recent GCC/LLVM

* **Platform**:

Rocky 8

* **MythTV version**:

master

* **Package version**:

N/A

* **Component**:

core

### What steps will reproduce the bug?

Compile MythTV from source on Rocky 8 with system clang-13, or software collections gcc-11

### How often does it reproduce? Is there a required condition?

Every time

### What is the expected behaviour?

Compile does not fail

### Additional information

When attempting to compile on Rocky 8 using clang I encountered a failure to build with the following (minimally extracted) error message:
```

  In file included from mythbinaryplist.cpp:34:
  In file included from /usr/include/qt5/QtCore/QtEndian:1:
  In file included from /usr/include/qt5/QtCore/qendian.h:44:
  /usr/include/qt5/QtCore/qfloat16.h:300:7: error: explicit specialization of undeclared template class 'numeric_limits'
  class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
           ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ...
```

This was reported upstream against Qt 5.15.2 and is fixed in Qt 5.15.9 (which is not yet publicly released, although the KDE project has made available backports for those on the available Qt 5.15 LTS release series (but not all distros / releases apply those backports, or apply them in a timely fashion)), and recent compilers).

The Qt bug report is: https://bugreports.qt.io/browse/QTBUG-90395 which is a result of the changes documented in https://gcc.gnu.org/gcc-11/porting_to.html

The specific case where I encountered the issue was a test build on Rocky 8 with the system clang compiler which is LLVM 13, and the software collections GCC 11.  This is, admittedly, somewhat of an edge case, as one needs to be using a distro that has Qt5 but has not applied the KDE backports, and using a recent compiler, but such cases can and do exist.

The mitigation (as discussed in the Qt bug report) is trivial for applications to implement (add a "#include \<limits>" before the "#include \<QtEndian>"), although it is also not ideal for applications to have to do so.

A PR for master will be created to address the issue (it will be desirable to be backported into fixes/32 after some settling time if merged to master).

**Comment by ulmus-scott** [comment]
I know it might take longer to fix the issue, but isn't this a bug in Rocky Linux's Qt and shouldn't it be be fixed there?

Regardless, adding comments to the includes to reference this issue would be a good idea, since the includes are not directly used (and in the wrong order).  Something like: `// workaround QTBUG-90395 see https://github.com/MythTV/mythtv/issues/631`.

**Comment by garybuhrmaster** [comment]
Closed with merge of PR #632
